obal/data/module_utils/obal.py

```diff
@@ -21,19 +38,19 @@ def specfile_macro_lookup(specfile, macro_str, scl=None, dist=None, macros=None)
         specfile
     ]
 
-    if dist:
-        command += ['--define', '"dist %s"' % dist]
-    else:
-        command += ['--undefine', 'dist']
+    return macro_lookup(command, scl=scl, dist=dist, macros=macros)
 
-    if scl:
-        command += ['--define', '"scl %s"' % scl]
 
-    if macros is not None:
-        for (macro, value) in macros.items():
-            command += ['--define', '"%s %s"' % (macro, value)]
+def srpm_macro_lookup(srpm, macro_str, scl=None, dist=None, macros=None):
+    """expand a given macro from an srpm"""
+    command = [
+        'rpmquery',
+        '--queryformat',
+        macro_str,
+        srpm
```

Comment:
```suggestion
        '--package',
        srpm
```

This will fix the exit code 1 on Ubuntus old RPM. (While still work on Fedora ;-)